packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -69,11 +83,13 @@ export async function buildSwingset(
   { debugName = undefined, slogCallbacks, slogSender },
 ) {
   const debugPrefix = debugName === undefined ? '' : `${debugName}:`;
-  /** @type {import('@agoric/swingset-vat').SwingSetConfig | null} */
-  let config = await loadSwingsetConfigFile(vatconfig);
-  if (config === null) {
-    config = loadBasedir(vatconfig);
+  let swingsetConfig = await loadSwingsetConfigFile(vatconfig);
```

Comment:
might be appropriate to call this one `config` as the union one, then define `swingSetConfig` as the leftovers once you remove the three `CosmicSwingsetConfig` properties, and pass `swingSetConfig` into swingset so it never sees the extras

Reply:
actually given the number of references to `config`, maybe this combined one should be called `combinedConfig`, and then we quickly split it into `{ coreProposals, clearStorageSubstrees, exportStorageSubtrees, ...config }`